rs/ic_os/os_tools/guest_vm_runner/src/main.rs

```diff
@@ -180,33 +180,18 @@ impl VirtualMachine {
         })
     }
 
-    fn try_destroy_existing_vm(libvirt_connect: &Connect, vm_domain_name: &str) -> Result<()> {
+    fn try_destroy_existing_vm(libvirt_connect: &Connect, vm_domain_name: &str) {
         if let Ok(existing_domain) = Domain::lookup_by_name(libvirt_connect, vm_domain_name) {
             eprintln!("Attempting to destroy existing '{vm_domain_name}' domain");
-            existing_domain
+            let _ = existing_domain
                 .destroy_flags(VIR_DOMAIN_DESTROY_GRACEFUL)
-                .context("Failed to destroy existing domain")?;
-            eprintln!("Successfully destroyed existing domain");
+                .inspect_err(|err| eprintln!("destroy_flags failed: {err}"));
+            let _ = existing_domain
+                .undefine()
+                .inspect_err(|err| eprintln!("undefine failed: {err}"));
```

Comment:
nit: It might be more clear to avoid the `must_use` with an `if let`, since we throw the result away anyway, but that may work out to an extra line on each. Up to you 🤷‍♀️.